File: tests/fixtures/real_world/cpp/cfg/lambda_taint.cpp
#include <cstdlib>
#include <string>
#include <algorithm>
#include <vector>

void handle(const char* user_input) {
    std::string cmd(user_input);
    std::vector<int> v = {1, 2, 3};
    std::for_each(v.begin(), v.end(), [](int x) {
        if (x == 0) return; // lambda-local return
    });
    // This sink should still be reachable
    system(cmd.c_str());
}

File: tests/fixtures/real_world/java/cfg/lambda_taint.java
import javax.servlet.http.*;

public class LambdaTaint {
    public void handle(HttpServletRequest req, HttpServletResponse resp) throws Exception {
        String input = req.getParameter("cmd");
        // Lambda should isolate taint — return inside lambda should NOT
        // kill the parent function's control flow
        java.util.Arrays.asList("a").forEach(x -> {
            if (x.equals("skip")) return; // lambda-local return
        });
        // This sink should still be reachable
        Runtime.getRuntime().exec(input);
    }
}
